### DISCUSSION TITLE
Need help understanding "Extra Device" delta calculation

### DISCUSSION THREAD

**Discussion by jsphvrghs** [artisan-roaster-scope/artisan#1123] [Q&A] [answered]
Hello,
I'm using Artisan 2.8.2

I'm trying to get RoR for ET from an external source via a websocket.

So far I've managed to get the ET values into artisan-scope and I am able to plot the ET curve.

Where I've run into a wall is the calculation of Delta ET and ET-BT

I tried the instructions from [here](https://artisan-roasterscope.blogspot.com/2015/10/signals-symbolic-assignments-and-plotter.html) and [here](https://artisan-roasterscope.blogspot.com/2019/11/symbolic-formulas-basics-new-variables.html)

I can't seem to get the Delta ET curve and the LCD is also always showing 0.0
For ET-BT I padded some additional value to avoid messing with my axes settings for now. But the value in the LCD is always 1.0 and there is no curve.

My settings are as below:

[artisan-settings_ET_RoR.txt](https://github.com/artisan-roaster-scope/artisan/files/10943728/artisan-settings_ET_RoR.txt)

Is there something I'm missing or setup incorrectly? 

Thanks for your help in advance.

**Comment by bigl00z3** [comment, marked as answer]
Not totally sure I understand.
I had a look at your settings. Why don't you use the built in RoR calculation?

In your `Config/Extra devices`, `ET` is in position as `Y9`
![image](https://user-images.githubusercontent.com/8886922/224373786-ef1afe65-25ea-4119-a831-abad23e0a23a.png)

In `Symb ET/BT`, you read `ET` as `Y3` which is your `Bean Temperature` in extra device:
![image](https://user-images.githubusercontent.com/8886922/224374195-919de0d3-c499-4075-9563-90304947f5ca.png)

If I understand your settings correctly (maybe I didn't), I would have:

- `ET` referring to `Y9` (1st device on line 4 of extra devices)
- `BT` referring to `Y3` (1st device on line 1 of extra devices)
![image](https://user-images.githubusercontent.com/8886922/224374872-238f628e-6e7e-40a9-9be4-555bcd08dc03.png)

and I would tick the ET boxes in the first tab:
![image](https://user-images.githubusercontent.com/8886922/224375238-2796efa2-044a-4867-abf8-fbbf59692ef0.png)

then for RoR I would turn it on in `Config/Curves...`
![image](https://user-images.githubusercontent.com/8886922/224375771-143424b3-f34a-4458-8698-d890c723c8c0.png)

Nico

**Reply by jsphvrghs** [reply]
Woah! How did you figure out that it was Y9? I tried everything from Y0 to Y4 and then gave up. 

I am getting sensible values now.

**Reply by bigl00z3** [reply]
if you click on `Help `you have the explanations:

- Y3 	Extra # 1 T1 value
- Y4	Extra # 1 T2 value
- Y5	Extra # 2 T1 value
- Y6	Extra # 2 T2 value
- ... 	...and so forth

`Y1` and `Y2` are for `ET` and `BT `in the first device and then you continue with Extra Devices:
![image](https://user-images.githubusercontent.com/8886922/224393879-0558d4b0-ab73-4f46-9193-8a905eb2d673.png)

**Reply by jsphvrghs** [reply]
Aah! OK! Something new I learned today. Thanks a ton! Your simple and clear explanation saved my day! Thanks @bigl00z3
